cli/aws-cron-validator.ts

```diff
@@ -0,0 +1,109 @@
+
+    public static validate(expression: string): string {
+        if (!expression.trim()) {
+            throw new AWSCronError(
+                `No parameters entered, this format is required in UTC: 0 20 ? * SUN-FRI *`
+            );
+        }
+        const valueCount = expression.split(" ").length;
+        if (valueCount !== 6) {
+          throw new AWSCronError(
+              `Incorrect amount of parameters in '${expression}'. 6 required, ${valueCount} provided.`
+          );
+        }
+
+        const [minute, hour, dayOfMonth, month, dayOfWeek, year] = expression.split(" ");
+
+        // special handling for Day of Month and Day of Week
+        if (!((dayOfMonth === "?" && dayOfWeek !== "?") || (dayOfMonth !== "?" && dayOfWeek === "?"))) {
+            throw new AWSCronError(
+                `Invalid combination of day-of-month '${dayOfMonth}' and day-of-week '${dayOfWeek}'. One must be a question mark (?)`
+            );
+        }
+
+        if (!new RegExp(this.minuteRegex()).test(minute)) {
+            throw new AWSCronError(`Invalid minute value '${minute}'.`);
+        }
+        if (!new RegExp(this.hourRegex()).test(hour)) {
+            throw new AWSCronError(`Invalid hour value '${hour}'.`);
+        }
+        if (!new RegExp(this.dayOfMonthRegex()).test(dayOfMonth)) {
+            throw new AWSCronError(`Invalid day-of-month value '${dayOfMonth}'.`);
+        }
+        if (!new RegExp(this.monthRegex(), 'i').test(month)) {
+            throw new AWSCronError(`Invalid month value '${month}'.`);
+        }
+        if (!new RegExp(this.dayOfWeekRegex(), 'i').test(dayOfWeek)) {
+            throw new AWSCronError(`Invalid day-of-week value '${dayOfWeek}'.`);
+        }
+        if (!new RegExp(this.yearRegex()).test(year)) {
+            throw new AWSCronError(`Invalid year value '${year}'.`);
+        }
+
+        return expression;
+    }
+
+    // Build up regex's incrementally improving readability of the above expressions
+    private static rangeRegex(values: string): string {
+        return `(${values}|(\\*\\-${values})|(${values}\\-${values})|(${values}\\-\\*))`;
+    }
+
+    private static listRangeRegex(values: string): string {
+        const range = this.rangeRegex(values);
+        return `(${range}(\\,${range})*)`;
+    }
+
+    private static slashRegex(values: string): string {
+        const range = this.rangeRegex(values);
+        return `((\\*|${range}|${values})\\/${this.numbers})`;
+    }
+
+    private static listSlashRegex(values: string): string {
+        const slash = this.slashRegex(values);
+        const slashOrRange = `(${slash}|${this.rangeRegex(values)})`;
+        return `(${slashOrRange}(\\,${slashOrRange})*)`;
+    }
+
+    private static commonRegex(values: string): string {
+        return `(${this.listRangeRegex(values)}|\\*|${this.listSlashRegex(values)})`;
+    }
+
+    private static minuteRegex(): string {
+        return `^(${this.commonRegex(this.minuteExp)})$`;
```

Comment:
nit: Given that the const `minuteExp` isn't used anywhere other than here, what have you separated it out to be its own constant? I see some of them are re-used but the ones that aren't might be more readable if you did them inline.